backend/YouTubeV2.Api.Tests/WebApplicationFactoryExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace YouTubeV2.Api.Tests
+{
+    internal static class WebApplicationFactoryExtensions
+    {
+        internal static async Task DoWithinScope<TService>(
+            this WebApplicationFactory<Program> webApplicationFactory, Func<TService, Task> action)
+        where TService : notnull
+        {
+            using var serviceScope = webApplicationFactory.Services.GetRequiredService<IServiceScopeFactory>().CreateScope();
+            var service = serviceScope.ServiceProvider.GetRequiredService<TService>();
+            await action(service);
+        }
+
+        internal static async Task DoWithinScope<TService, UService>(
```

Comment:
maybe think a little bit about a way how to merge this two DoWithinScope methods idk if there is a way to do it